src/storage/txn/actions/prewrite.rs

```diff
@@ -446,6 +446,16 @@ impl<'a> PrewriteMutation<'a> {
             .into());
         }
 
+        // A key can be flushed multiple times for a Pipelined-DML transaction.
+        // A latter flush with `should_not_exist` should return error if a previous
+        // flush of the key writes a value
+        if self.should_not_exist && matches!(lock.lock_type, LockType::Put) {
```

Comment:
Is it possible to check if the lock is a `pipelined lock` type? It seems the check is unnecessary for normal transctions.

Reply:
Makes sense